internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -230,7 +231,7 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 
 	if swag.StringValue(cluster.Kind) == models.ClusterKindCluster && !clusterInstall.Spec.HoldInstallation {
 		// Day 1
-		pullSecret, err := getAndLabelPullSecret(ctx, r.Client, r.APIReader, clusterDeployment.Spec.PullSecretRef, req.Namespace)
+		pullSecret, err := r.PullSecretClient.GetValidPullSecret(ctx, getPullSecretKey(req.Namespace, clusterDeployment.Spec.PullSecretRef))
 		if err != nil {
 			log.WithError(err).Error("failed to get pull secret")
```

Comment:
It can also be an invalid secret if we get here. Can we differentiate between a failure the fetch the secret and a case of an invalid secret?  So we could treat an invalid secret as we do for invalid release image: https://github.com/openshift/assisted-service/blob/a5cafb300700e5440504eae1de1cdb4ee0884619/internal/controller/controllers/clusterdeployments_controller.go#L387

Reply:
The result is the same: the secret can not be use and the user should check its configurations. I did not change the API dramatically in this PR because it suppose to be a small change

Reply:
Then maybe rephrase the log message to cover both cases, e.g. "missing or invalid Pull Secret"

Reply:
After investigating the code there is no need to change the log. There are ample logs in ValidatePullSecret function that describe specifically what went wrong. Then, in addition, there is this logs which is a legacy log. Together, they provide an accurate picture of what is wrong and how to fix it.